src/vcpkg/platform-expression.cpp

```diff
@@ -139,26 +142,114 @@ namespace vcpkg::PlatformExpression
+
+                    case ExprKind::op_empty: return result;
+
+                    default:
+                        // op_identifier and op_invalid both indicate a syntax error, which should have
+                        // already been flagged by expr_operator.
+                        return result;
+                }
+            }
+
+            // platform-expression-and =
+            // | platform-expression-not, { "&", optional-whitespace, platform-expression-not } ;
+            // | platform-expression-not, { "and", optional-whitespace, platform-expression-not } ;
+            //
+            // platform-expression-or =
+            // | platform-expression-not, { "|", optional-whitespace, platform-expression-not } ;
+            //
+            // platform-expression-low-precedence-or =
+            // | platform-expression-not, { ",", optional-whitespace, platform-expression-not } ;
+            ExprKind expr_operator()
+            {
+                auto oper = cur();
+
+                // Support chains of the vcpkg operators (`&`, `|`)  to avoid breaking backwards compatibility
+                switch (oper)
+                {
+                    case '|':
+                    case '&':
+                        do
+                        {
+                            next();
+                        } while (allow_multiple_binary_operators() && cur() == oper);
+                        break;
+                }
+
+                switch (oper)
                 {
                     case '|':
                     {
                         // { "|", optional-whitespace, platform-expression-not }
-                        return expr_binary<'|', '&'>(std::make_unique<ExprImpl>(ExprKind::op_or, std::move(result)));
+                        return ExprKind::op_or;
                     }
                     case '&':
                     {
                         // { "&", optional-whitespace, platform-expression-not }
-                        return expr_binary<'&', '|'>(std::make_unique<ExprImpl>(ExprKind::op_and, std::move(result)));
+                        return ExprKind::op_and;
+                    }
+                    case ',':
+                    {
+                        // { ",", optional-whitespace, platform-expression-not }
+                        // "," is a near-synonym of "|", with the differences that it can be combined with "&"/"and",
+                        // but has lower precedence
+                        next();
+                        return ExprKind::op_low_precedence_or;
+                    }
+                    case 'a':
+                    {
+                        // { "and", optional-whitespace, platform-expression-not }
+                        // "and" is a synonym of "&"
+                        std::string name = match_zero_or_more(is_identifier_char).to_string();
+                        if (!name.empty() && (name == "and"))
```

Comment:
```suggestion
                        if (name == "and")
```
`== "and"` implies `!empty()`